File: Folder Python/Film.py
# deklarasi class
class Film:
    # private atribut
    def __init__(self, id_film, judul, genre, durasi, sutradara):
        self.__id_film = id_film  # inisialisasi
        self.__judul = judul  # inisialisasi
        self.__genre = genre  # inisialisasi
        self.__durasi = durasi  # inisialisasi
        self.__sutradara = sutradara  # inisialisasi

    # getter
    def get_id_film(self):
        return self.__id_film  # mengambil value

    def get_judul(self):
        return self.__judul  # mengambil value

    def get_genre(self):
        return self.__genre  # mengambil value

    def get_durasi(self):
        return self.__durasi  # mengambil value

    def get_sutradara(self):
        return self.__sutradara  # mengambil value

    # prosedur untuk mengupdate data
    def update(self, judul_baru, genre_baru, durasi_baru, sutradara_baru):
        self.__judul = judul_baru
        self.__genre = genre_baru
        self.__durasi = durasi_baru
        self.__sutradara = sutradara_baru

    # prosedur untuk menampilkan data
    def tampilkan_data(self):
        print("ID Film    :", self.get_id_film())
        print("Judul      :", self.get_judul())
        print("Genre      :", self.get_genre())
        print("Durasi     :", self.get_durasi(), "menit")
        print("Sutradara  :", self.get_sutradara())

File: Folder PHP/Film.php
<?php

class Film
{
    private string $id_film;
    private string $judul;
    private string $genre;
    private int $durasi;
    private string $sutradara;
    private ?string $gambar;

    // Constructor
    public function __construct(
        string $id,
        string $judul,
        string $genre,
        int $durasi,
        string $sutradara,
        ?string $gambar = null
    ) {
        $this->id_film = $id;
        $this->judul = $judul;
        $this->genre = $genre;
        $this->durasi = $durasi;
        $this->sutradara = $sutradara;
        $this->gambar = $gambar;
    }

    // Getter
    public function getId(): string
    {
        return $this->id_film;
    }

    public function getJudul(): string
    {
        return $this->judul;
    }

    public function getGenre(): string
    {
        return $this->genre;
    }

    public function getDurasi(): int
    {
        return $this->durasi;
    }

    public function getSutradara(): string
    {
        return $this->sutradara;
    }

    public function getGambar(): ?string
    {
        return $this->gambar;
    }

    // Setter
    public function setId(string $id_film): void
    {
        $this->id_film = $id_film;
    }

    public function setJudul(string $judul): void
    {
        $this->judul = $judul;
    }

    public function setGenre(string $genre): void
    {
        $this->genre = $genre;
    }

    public function setDurasi(int $durasi): void
    {
        if ($durasi >= 0) {
            $this->durasi = $durasi;
        }
    }

    public function setSutradara(string $sutradara): void
    {
        $this->sutradara = $sutradara;
    }

    public function setGambar(?string $gambar): void
    {
        $this->gambar = $gambar;
    }

    // Menampilkan data
    public function tampilkanData(): void
    {
        echo "ID Film: " . htmlspecialchars($this->id_film) . "<br>";
        echo "Judul: " . htmlspecialchars($this->judul) . "<br>";
        echo "Genre: " . htmlspecialchars($this->genre) . "<br>";
        echo "Durasi: " . $this->durasi . " menit<br>";
        echo "Sutradara: " . htmlspecialchars($this->sutradara) . "<br>";
        echo "Gambar: " . htmlspecialchars($this->gambar ?? "Tidak ada") . "<br>";
    }
}
?>